modules/javafx.graphics/src/main/java/javafx/stage/Stage.java

```diff
@@ -1213,6 +1217,88 @@ public void toBack() {
+     * optionally adjusted to keep the stage within screen edge constraints.
+     * <p>
+     * The {@code anchorPolicy} controls whether an alternative anchor may be used when the preferred anchor
+     * would violate screen edge constraints. Depending on the policy, the preferred anchor location may be
+     * mirrored to the other side of the window horizontally or vertically, or an anchor might be selected
+     * automatically. If no alternative anchor yields a better placement, the specified {@code anchor} is used.
+     * <p>
+     * The {@code screenPadding} parameter defines per-edge constraints against the current screen bounds.
+     * Each inset value specifies the minimum distance to maintain between the stage edge and the corresponding
+     * screen edge. A value {@code >= 0} enables the corresponding edge constraint; a negative value disables
+     * the constraint for that edge. Enabled constraints effectively shrink the usable screen area by the
+     * given insets. For example, a left inset of {@code 10} ensures the stage will not be placed closer than
+     * 10 pixels to the left screen edge.
+     * <p>
+     * This method may be called either before or after {@link #show()} or {@link #showAndWait()}.
+     * If called <em>before</em> the stage is shown, then
+     * <ol>
+     *     <li>any previous call to {@link #centerOnScreen()} is disregarded,
+     *     <li>the {@link #xProperty() X} and {@link #yProperty() Y} properties are not updated
+     *         immediately; instead, they are updated after the stage is shown.
+     * </ol>
+     *
+     * @param anchorX the requested horizontal location of the anchor point on the screen
+     * @param anchorY the requested vertical location of the anchor point on the screen
+     * @param anchor the point on the stage that should coincide with {@code (anchorX, anchorY)} on the screen
+     * @param anchorPolicy controls alternative anchor selection if the preferred placement violates
+     *                     enabled screen constraints
+     * @param screenPadding per-edge minimum distance constraints relative to the screen edges;
+     *                      values {@code < 0} disable the constraint for the respective edge
+     * @throws NullPointerException if {@code anchor}, {@code anchorPolicy}, or {@code screenPadding} is {@code null}
+     * @since 26
+     */
+    public final void relocate(double anchorX, double anchorY, AnchorPoint anchor,
+                               AnchorPolicy anchorPolicy, Insets screenPadding) {
+        var request = WindowBoundsUtil.newDeferredRelocation(anchorX, anchorY, anchor, anchorPolicy, screenPadding);
+
+        if (isShowing()) {
+            request.accept(this);
+        } else {
+            this.relocationRequest = request;
+        }
+    }
+
+    @Override
+    public void centerOnScreen() {
+        relocationRequest = null; // cancel previous relocation request
```

Comment:
As far as I understand, with the new anchor system, a center on screen can easily be achieved as well? I wonder if we could also deprecate this method instead. But not sure if I understand it correctly.

Reply:
No, the anchor is always a point _on the window being shown_, not on the screen. All anchor APIs require you to specify an explicit screen location, and the window is then moved so that the anchor coincides with that screen location. So to emulate `centerOnScreen()`, you'd have to manually get the screen bounds, and then calculate the center.

Reply:
I like it, because we can easily cover almost all the possible cases.  

Should we also allow to specify the `Screen`?  In a third parameter to this extended method?  And if the `Screen` is `null`, the logic would use the default screen or the owner's one?  What do you think?

Reply:
I've added another overload that accepts a `Screen`. The overloads _without_ the screen parameter will always refer to the _current_ screen. This allows users to relocate a stage on its current screen as if it was the primary screen (i.e. with `(0, 0)` being the top-left corner).